### ISSUE TITLE
Crash for run 355207-355208 in express StreamHLTMonitor

### ISSUE THREAD

**Issue by ttedeschi** [cms-sw/cmssw#38626]
As pointed out here https://cms-talk.web.cern.ch/t/paused-jobs-for-run-355207-collisions/12561 the following error is encountered when running ```express_StreamHLTMonitor``` workflow for both runs 355207 and 255208

```
2022-07-07 09:05:58,824:CRITICAL:CMSSW:Error running cmsRun
{'arguments': ['/bin/bash', '/srv/job/WMTaskSpace/cmsRun1/cmsRun1-main.sh', '', 'slc7_amd64_gcc10', 'scramv1', 'CMSSW', 'CMSSW_12_3_6', 'FrameworkJobReport.xml', 'cmsRun', 'PSet.py', '']}
CMSSW Return code: 8006

2022-07-07 09:05:58,824:CRITICAL:CMSSW:Error message: An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 355207 lumi: 582 event: 669724552 stream: 6
   [1] Running path 'dqmoffline_step'
   [2] Calling method for module BTVHLTOfflineSource/'BTVHLTOfflineSource'
Exception Message:
RefCore: A request to resolve a reference to a product of type 'reco::Candidate' with ProductID '1:2358'
can not be satisfied because the product cannot be found.
Probably the branch containing the product is not stored in the input file.
   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.
```

full info can be found here:
```/afs/cern.ch/user/c/cmst0/public/PausedJobs/ExpressHLTMonitor/job/WMTaskSpace/cmsRun1```

a possible fix, even a temporary one, should be found urgently

**Comment by jfernan2** [comment]
FYI @marco-link @johnalison @JyothsnaKomaragiri @natalia-korneeva @SWuchterl as BTV HLT DQM And Validation Code Developers 
https://twiki.cern.ch/twiki/bin/viewauth/CMS/DQMContacts#Btag_and_vertexing

**Comment by NiclasEich** [comment]
BTV HLT DQM developers (@marco-link @terrill37 me) are working on it and can replicate the error.

**Comment by mmusich** [comment]
seems the place in which it fails is at

https://github.com/cms-sw/cmssw/blob/1157e3664510ba0f9300604d3e85dab99771ffbe/DQMOffline/Trigger/plugins/BTVHLTOfflineSource.cc#L798

**Comment by NiclasEich** [comment]
> seems the place in which it fails is at
> 
> https://github.com/cms-sw/cmssw/blob/1157e3664510ba0f9300604d3e85dab99771ffbe/DQMOffline/Trigger/plugins/BTVHLTOfflineSource.cc#L798

Indeed, we have implemented a sanity check for null-pointer in the following `push_back` calls, which resolves the error.

We are currently preparing the PRs and checking the tests.

**Comment by missirol** [comment]
Thanks to the people who are debugging.

I'm in favour of a(ny) fix being integrated asap, but it would also be useful to know the reason behind this failure. Is some track collection missing in the input file wrt what the DQM client needs/expects?

Knowing better the reason behind the problem might suggest a workaround that we could apply to the HLT menu to be used online this weekend. This could reduce pressure to deploy a new release online.

**Comment by marco-link** [comment]
Opened the PR.
Thanks @mmusich for working on this in parallel (you were ~20 minutes ahead of us :grinning:)

I'll prepare backports for 12_3_X and 12_4_X.

**Comment by soureek** [comment]
@mmusich, @tvami, and @missirol  This issue is resolved long time ago. Please close it.

**Comment by mmusich** [comment]
@soureek it would be better if @cms-sw/dqm-l2 sign it first.
